crates/adapters/src/format/avro/deserializer.rs

```diff
@@ -0,0 +1,1596 @@
+//! Deserializer backed by the Avro `Value` type, adapted from the `apache-avro`
+//! crate with several modifications:
+//!
+//! * Convert time and timestamp values from microseconds to milliseconds, as
+//!   expected by our `Timestamp` and `Time` types.
+//! * Deserializes optional types from non-nullable schemas (normally
+//!   `Option<T>` can only be serialized from an Avro value of the shape `["null", V]`).
+//! * Deserializes non-optional types from nullable schemas.
+
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Did you modify some rust code, or some Java code?
If it's Java I don't know if you need this license.

Reply:
It's Rust.